src/adapters/unix_socket.rs

```diff
@@ -0,0 +1,212 @@
+    ListeningInfo, PendingStatus,
+};
+use crate::network::{RemoteAddr, Readiness, TransportConnect, TransportListen};
+
+use mio::event::{Source};
+use mio::net::{UnixListener, UnixStream};
+
+use std::mem::MaybeUninit;
+use std::net::{IpAddr, Ipv4Addr, SocketAddr};
+use std::io::{self, ErrorKind, Read, Write};
+use std::ops::Deref;
+use std::path::{Path, PathBuf};
+
+// Note: net.core.rmem_max = 212992 by default on linux systems
+// not used because w euse unixstream I think?
+// TODO: delete this if I PR
+pub const MAX_PAYLOAD_LEN: usize = 212992;
+
+/// From tcp.rs
+/// Size of the internal reading buffer.
+/// It implies that at most the generated [`crate::network::NetEvent::Message`]
+/// will contains a chunk of data of this value.
+pub const INPUT_BUFFER_SIZE: usize = u16::MAX as usize; // 2^16 - 1
+
+// We don't use the SocketAddr, we just striaght up get the path from config.
+pub fn create_null_socketaddr() -> SocketAddr {
+    SocketAddr::new(IpAddr::V4(Ipv4Addr::new(0,0,0,0)), 0)
+}
+
+#[derive(Clone, PartialEq, Eq, Hash, Debug)]
+pub struct UnixSocketListenConfig {
+    path: PathBuf,
+}
+
+impl Default for UnixSocketListenConfig {
+    fn default() -> Self {
+        // TODO: better idea? I could make this into an option later and complain if empty.
+        Self { path: "/tmp/mio.sock".into() }
+    }
+}
+
+#[derive(Clone, PartialEq, Eq, Hash, Debug)]
+pub struct UnixSocketConnectConfig {
+    path: PathBuf,
+}
+
+impl Default for UnixSocketConnectConfig {
+    fn default() -> Self {
+        // TODO: better idea? I could make this into an option later and complain if empty.
+        Self { path: "/tmp/mio.sock".into() }
+    }
+}
+
+pub(crate) struct UnixSocketAdapter;
+impl Adapter for UnixSocketAdapter {
+    type Remote = RemoteResource;
+    type Local = LocalResource;
+}
+
+pub(crate) struct RemoteResource {
+    stream: UnixStream
+}
+
+impl Resource for RemoteResource {
+    fn source(&mut self) -> &mut dyn Source {
+        &mut self.stream
+    }
+}
+
+// taken from tcp impl
+pub fn check_stream_ready(stream: &UnixStream) -> PendingStatus{
+    if let Ok(Some(_)) = stream.take_error() {
+        return PendingStatus::Disconnected;
+    } 
+
+    return PendingStatus::Ready;
+}
+
+impl Remote for RemoteResource {
+    fn connect_with(
+        config: TransportConnect,
+        remote_addr: RemoteAddr,
+    ) -> io::Result<ConnectionInfo<Self>> {
+
+        let stream_config = match config {
+            TransportConnect::UnixSocket(config) => config,
+            _ => panic!("Internal error: Got wrong config"),
+        };
+        
+        match UnixStream::connect(stream_config.path) {
+            Ok(stream) => {
+                Ok(ConnectionInfo {
+                    remote: Self {
+                        stream
+                    },
+                    // the unixstream uses SocketAddr from mio that can't be converted
```

Comment:
Out of curiosity, what then identifies an unix socket?

Reply:
local_addr and peer_addr are currently a placeholder value, the unix sockets seem to have a totally different local and peer addr format.